core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -197,10 +197,10 @@
                     DEFAULT_DEFAULT_CLIENT_IDLE_TIMEOUT_MILLIS,
                     value -> value >= 0);
 
-    private static final long DEFAULT_DEFAULT_HTTP2_PING_TIMEOUT_MILLIS = 0; // Disabled
-    private static final long DEFAULT_HTTP2_PING_TIMEOUT_MILLIS =
-            getLong("defaultHttp2PingTimeoutMillis",
-                    DEFAULT_DEFAULT_HTTP2_PING_TIMEOUT_MILLIS,
+    private static final long DEFAULT_DEFAULT_PING_INTERVAL_MILLIS = 0; // Disabled
```

Comment:
What would be the sensible default for this flag?

Reply:
Hmm... I thought `PING` is an optional feature. If we turn it on by default, what about `5000` milliseconds?
Because the default PING interval should be less than the default idle timeout.
* the default idle timeout of service-side is `15000` milliseconds
* the default idle timeout of client-side is `10000` milliseconds

Reply:
Sounds good to me. I guess it'll be nice if we do not send ping frames too often so that some HTTP/2 servers reject us, e.g. gRPC-Java has such check. Could you double check? Related issue: #2639

Reply:
https://github.com/grpc/proposal/blob/master/A8-client-side-keepalive.md

The default keepalive configuration of upstream gRPC server seems to be conservative.  
* The default value of `permitKeepAliveTime` is 5 minutes.
* `permitKeepAliveWithoutCalls()` is disable by default.


These means 
* Our gRPC client should not send a PING frame with no outstand streams
* The PING interval should be longer than 5 minutes with active streams.
 
Maybe we need a different default keepalive configuration for gRPC? Any thoughts? @anuraaga

Reply:
I guess we need to be as conservative as gRPC-Java by default to prevent any surprises. `KeepAliveEnforcer` contains the core logic of the defense mechanism: https://github.com/grpc/grpc-java/blob/master/netty/src/main/java/io/grpc/netty/KeepAliveEnforcer.java which seems to be:
- interested only in the interval between PING frames. It doesn't take other traffic into account.
  - Perhaps we need another parameter, like 'minimum interval between pings'?
- accepting up to 2 violations (dunno why)

Reply:
I'm not sure this is useful for us, but gRPC client limits the min ping interval.
https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/internal/KeepAliveManager.java#L235-L240